web/src/app/services/loi/loi.service.spec.ts

```diff
@@ -50,4 +54,38 @@ describe('LocationOfInterestService', () => {
     );
     expect(service).toBeTruthy();
   });
+
+  it('should get inferred name from loi property', () => {
+    const service: LocationOfInterestService = TestBed.inject(
+      LocationOfInterestService
+    );
+
+    const getMockLoi = (properties: ImmutableMap<string, string | number>) => {
+      return new GenericLocationOfInterest(
+        'loi001',
+        'job001',
+        new Point(new Coordinate(0.0, 0.0)),
+        properties
+      );
+    };
+
+    // Empty or non applicable properties should not return an inferred name.
+    let properties: ImmutableMap<string, string | number> = ImmutableMap();
+    expect(service.getLoiNameFromProperties(getMockLoi(properties))).toBe(null);
+
+    // Use properties to choose name for loi.
+    properties = ImmutableMap([['id', 'loi1']]);
+    expect(service.getLoiNameFromProperties(getMockLoi(properties))).toBe(
+      'loi1'
+    );
+
+    // Two separate possible names exist, choose one with higher priority.
+    properties = ImmutableMap([
+      ['id', 'loi1'],
+      ['name', 'loi 1'],
+    ]);
+    expect(service.getLoiNameFromProperties(getMockLoi(properties))).toBe(
+      'loi 1'
+    );
```

Comment:
nit: This look like 3 test cases to me, consider separate them.

Suggestion:
You could create another `describe` here like:
```
describe('some name', () => {
  beforeEach(() => {
    const service: LocationOfInterestService = ...
    const getMockLoi = ...
  });
  it('Empty or non applicable properties should not return an inferred name', () => {
     blah blah blah
     expect(...
  });

  it('Use properties to choose name for loi, () => {
     blah blah blah
     expect(...
  });

  it('Two separate possible names exist, choose one with higher priority', () => {
     blah blah blah
     expect(...
  });
});
```